src/bench.js

```diff
@@ -227,63 +239,118 @@ table.sort((a, b) => {
 
 console.log(table.toString());
 
+
+// Benchnmark Utilities
+
+function instrument(parent) {
+  const {
+    appendChild,
+    insertBefore,
+    removeChild,
+    replaceChild
+  } = parent;
+  parent.operations = [];
+  parent.appendChild = function (newNode) {
+    this.operations.push(`appendChild(${newNode.textContent})`);
+    return appendChild.call(this, newNode);
+  };
+  parent.insertBefore = function (newNode, oldNode) {
+    this.operations.push(
+      oldNode ?
+        `insertBefore(${newNode.textContent}, ${oldNode.textContent})` :
+        `insertBefore(${newNode.textContent})`
+    );
+    return insertBefore.call(this, newNode, oldNode);
+  };
+  parent.removeChild = function (oldNode) {
+    this.operations.push(`removeChild(${oldNode.textContent})`);
+    return removeChild.call(this, oldNode);
+  };
+  parent.replaceChild = function (newNode, oldNode) {
+    this.operations.push(
+      `delete#replaceChild(${newNode.textContent}, ${oldNode.textContent})`
+    );
+    this.operations.push(
+      `insert#replaceChild(${newNode.textContent}, ${oldNode.textContent})`
+    );
+    return replaceChild.call(this, newNode, oldNode);
+  };
+}
+
+function reset(parent) {
+  parent.operations.splice(0);
+}
+
 function round(num) {
   return Math.round((num + Number.EPSILON) * 10) / 10;
 }
 
-function random(parent, diff) {
-  return diff(
-    parent,
-    parent.childNodes,
-    shuffleSeed.map((newIdx) => parent.childNodes[newIdx]),
-    parent.lastElementChild
-  );
+function verifyNodes(parent, childNodes, expected) {
+  return childNodes.length === expected &&
+          childNodes.every((row, i) => row === parent.childNodes[i]) &&
+          parent.childNodes.length === expected + (before ? 1 : 0);
 }
 
-function reverse(parent, diff) {
+
+// Benchnmark Functions
+
+function random(parent, diff, oldNodes) {
   return diff(
     parent,
-    parent.childNodes,
-    Array.from(parent.childNodes).reverse(),
-    parent.lastElementChild
+    oldNodes,
+    shuffleSeed.map((newIdx) => oldNodes[newIdx]),
+    before
```

Comment:
This `before` is not passed as argument - is that needed here?

Reply:
It's on top of the file hence needed in every diff call. You can remove the comment and see it used